File: src/main/java/beam/transEnergySim/agents/VehicleAgent.java
package beam.transEnergySim.agents;

import org.matsim.api.core.v01.Id;
import org.matsim.api.core.v01.population.Person;

import beam.transEnergySim.vehicles.api.VehicleWithBattery;

public interface VehicleAgent {

	Id<Person> getPersonId();
	VehicleWithBattery getVehicle();
}

File: src/main/scala/beam/agentsim/agents/vehicles/VehicleAgent.scala
package beam.agentsim.agents.vehicles

import beam.agentsim.agents.BeamAgent
import beam.agentsim.agents.BeamAgent.BeamAgentData
import beam.agentsim.agents.vehicles.VehicleAgent.VehicleData
import org.matsim.api.core.v01.Id


object VehicleAgent{
  object VehicleData{}

  case class VehicleData() extends BeamAgentData
}



/**
  *
  */
class VehicleAgent(override val id: Id[VehicleAgent], override val data: VehicleData) extends BeamAgent[VehicleData]{

}

